### PR TITLE
feat(mls-keypackages): add keypackageCountUseCase (AR-1846)

### PR DESCRIPTION
**Pull request by mchenani**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?
[AR-1846]

#### Test Coverage (Optional)
- [x] I have added automated test to this contribution
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

[AR-1846]: https://wearezeta.atlassian.net/browse/AR-1846

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/ClientScope.kt [hunk 1]
```diff
@@ -30,4 +32,6 @@ class ClientScope(
     val needsToRegisterClient: NeedsToRegisterClientUseCase get() = NeedsToRegisterClientUseCaseImpl(clientRepository)
     val registerPushToken: RegisterTokenUseCase get() = RegisterTokenUseCaseImpl(clientRepository, notificationTokenRepository)
     val deregisterNativePushToken: DeregisterTokenUseCase get() = DeregisterTokenUseCaseImpl(clientRepository, notificationTokenRepository)
+    val mlsKeyPackageCountUseCase: MLSKeyPackageCountUseCase
```
Comment (vitorhugods) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/ClientScope.kt:35]: Should this be public?

Maybe I'm missing some context. What's the usage of this use-case for the apps? I wonder if apps will ever need this. It seems like something Kalium will use internally only.

**Review by vitorhugods — COMMENTED** [review]
Code looks great. I may just lack a bit of context to understand its need.

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/keypackage/MLSKeyPackageCountUseCaseTest.kt [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+import io.mockative.verify
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.runTest
+import kotlin.test.BeforeTest
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertIs
+
+@OptIn(ExperimentalCoroutinesApi::class)
+class MLSKeyPackageCountUseCaseTest {
```
Comment (saleniuk) [inline logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/keypackage/MLSKeyPackageCountUseCaseTest.kt:25]: I'm not sure if it's necessary in this case but I think we agreed to create new test classes using "arrangement" pattern 🤔

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/ClientScope.kt (hunk 1 above)
Comment (mchenani) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/ClientScope.kt:35]: Yes, I used in the AR to show keyPackagesCount in the debug screen.

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/keypackage/MLSKeyPackageCountUseCase.kt [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+        })
+
+}
+
+
+sealed class MLSKeyPackageCountResult {
+    data class Success(val clientId: ClientId, val count: Int) : MLSKeyPackageCountResult()
+
+    sealed class Failure : MLSKeyPackageCountResult() {
+        class Generic(val genericFailure: CoreFailure) : Failure()
```
[the remark this answers is not in the document]
Comment (mchenani) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/keypackage/MLSKeyPackageCountUseCase.kt:34]: Right! My bad, will fix it. Thanks.

File: logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/keypackage/MLSKeyPackageCountUseCaseTest.kt (hunk 2 above)
Comment (mchenani) [reply logic/src/commonTest/kotlin/com/wire/kalium/logic/feature/keypackage/MLSKeyPackageCountUseCaseTest.kt:25]: Hmm! Right, but it's small test class. I think it's not necessary here.

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/ClientScope.kt (hunk 1 above)
Comment (vitorhugods) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/ClientScope.kt:35]: Nice.

Maybe we could have a `Debug` scope or something in the future. For things that should have less discoverability by consumers of Kalium.
Especially if we have some dangerous use cases for debug purposes 😆.

Comment (mchenani) [reply logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/ClientScope.kt:35]: Totally agree, that's the reason I'm asking for debug variant! it's hard to remember which parts we need to remove when we release the app.

**Pull request merged**